moved/src/move_execution/gas.rs

```diff
@@ -121,6 +121,33 @@ impl L1GasFee for EcotoneL1GasFee {
     }
 }
 
+/// Creates algorithm for calculating cost of publishing a transaction to layer-1 blockchain.
+pub trait CreateL1GasFee {
+    /// Extracts parameters from deposit transaction and creates the algorithm for calculating L1
+    /// gas cost.
+    fn for_deposit<'f, 'r>(&'f self, data: &'f [u8]) -> impl L1GasFee + 'r;
```

Comment:
I'm curious why we need the second lifetime parameter here. Seems like all the implementations have it resolve to `'static` and I can't think of a situation where we would have something other than that.

Reply:
I was able to use implicit lifetime and anonymouse lifetime combo for less verbose syntax.

Without the second lifetime we get the error:

```
error[E0505]: cannot move out of `transactions` because it is borrowed
   --> moved/src/state_actor.rs:242:45
    |
232 |         let mut transactions = transactions.peekable();
    |             ---------------- binding `transactions` declared here
233 |         let first_tx = transactions.peek();
    |                        ------------ borrow of `transactions` occurs here
...
242 |         for (tx_hash, tx, l1_cost_input) in transactions {
    |                                             ^^^^^^^^^^^^ move out of `transactions` occurs here
...
248 |                 l1_fee
    |                 ------ borrow later used here

```

I think this was the reason why I threw away the `peekable` which I originally used, glad you reminded me to put it back.

Reply:
This is the only spec I see that refers to first transaction being a deposit txn.
https://github.com/ethereum-optimism/optimistic-specs/blob/main/specs/deposits.md#kinds-of-deposited-transactions

We can just simply mention that based on the op-geth implementation the first transaction must be a deposit transaction.

Other than that I think this PR is good to merge ⛙